applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -74,6 +74,9 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
         if (GetGeometry().LocalSpaceDimension() == 1) {
             GetGeometry().DeterminantOfJacobian(det_J_container, this->GetIntegrationMethod());
             dN_dX_container = GetGeometry().ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
+            for (unsigned int index = 0; index < dN_dX_container.size(); ++index) {
+                dN_dX_container[index] /= det_J_container[index];
+            }
```

Comment:
Creating a version without unnecessary error messages and some shortening, the routine may look like the following to avoid the difference between 1D and 2D/3D:

    void CalculateLocalSystem(MatrixType&        rLeftHandSideMatrix,
                              VectorType&        rRightHandSideVector,
                              const ProcessInfo& rCurrentProcessInfo) override
    {
        KRATOS_TRY

        const GeometryType& r_geom = GetGeometry();
        const auto          number_of_integration_points =
            r_geom.IntegrationPointsNumber(this->GetIntegrationMethod());

        const GeometryType::ShapeFunctionsGradientsType& dN_de =
            r_geom.ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
        GeometryType::ShapeFunctionsGradientsType dN_dX_container(number_of_integration_points);
        Vector                                    det_J_container(number_of_integration_points);

        for (unsigned int pnt = 0; pnt < number_of_integration_points; pnt++) {
            Matrix J;
            GetGeometry().Jacobian(J, pnt, this->GetIntegrationMethod());
            Matrix Jinv;
            MathUtils<double>::GeneralizedInvertMatrix(J, Jinv, det_J_container[pnt]);
            dN_dX_container[pnt] = prod(dN_de[pnt], Jinv);
        }

        const auto integration_coefficients = CalculateIntegrationCoefficients(det_J_container);
        const auto conductivity_matrix =
            CalculateConductivityMatrix(dN_dX_container, integration_coefficients, rCurrentProcessInfo);
        const auto capacity_matrix = CalculateCapacityMatrix(integration_coefficients, rCurrentProcessInfo);

        AddContributionsToLhsMatrix(rLeftHandSideMatrix, conductivity_matrix, capacity_matrix,
                                    rCurrentProcessInfo[DT_TEMPERATURE_COEFFICIENT]);
        AddContributionsToRhsVector(rRightHandSideVector, conductivity_matrix, capacity_matrix);

        KRATOS_CATCH("")
    }